test_launch_ros/test/test_launch_ros/test_track_node_names.py

```diff
@@ -0,0 +1,161 @@
+
+
+def test_launch_node_with_name():
+    node = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 1
+
+
+def test_launch_node_without_name():
+    node = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_namespace='',
+        output='screen',
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 0
+
+
+def test_launch_composable_node_with_names():
+    node = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='composition',
+                node_plugin='composition::Listener',
+                node_name=TEST_NODE_NAME
+            )
+        ],
+        output='screen'
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 2
+
+
+def test_launch_composable_node_without_component_name():
+    node = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='composition',
+                node_plugin='composition::Listener',
+            )
+        ],
+        output='screen'
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 1
+    assert get_node_name_count(context, '/listener') == 1
+
+
+def test_launch_nodes_with_same_names():
+    node1 = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+
+    node2 = LifecycleNode(
+        package='lifecycle',
+        node_executable='lifecycle_listener',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+
+    node3 = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='composition',
+                node_plugin='composition::Listener',
+                node_name=TEST_NODE_NAME
+            )
+        ],
+        output='screen'
+    )
+
+    ld = LaunchDescription([node1, node2, node3])
+    context = _launch(ld)
```

Comment:
it would also be possible to capture the outuput here and check that the warning is being shown.

Reply:
same in other places were the count is bigger than one.

Reply:
How do we capture the logger output here?

Reply:
mmm, I rember having tried to use `pytest` to capture `stdout/stderr`, but it wasn't capturing the logger output.
You can give it a try and re-check if it works. If it doesn't work, disregard this comment.

Reply:
Does the contextlib capture not work for loggers? e.g.

```
from contextlib import redirect_stdout
import io

f = io.StringIO()
with redirect_stdout(f):
    some_stuff
s = f.getvalue()
```

Reply:
Done.

Reply:
> Thanks for the suggestion. I think this is what @ivanpauno is referring to, which looks a bit more idiomatic.

Yes, I was referring to that, sorry for not being too specific.